omxplayer/player.py

```diff
@@ -190,10 +190,15 @@ def monitor(self, process, on_exit):
                                    stdin=devnull,
                                    stdout=devnull,
                                    preexec_fn=os.setsid)
-        self._process_monitor = threading.Thread(target=monitor,
-                                                 args=(self, process, on_exit))
-        self._process_monitor.start()
-        return process
+        try:
+            self._process_monitor = threading.Thread(target=monitor,
+                                                     args=(self, process, on_exit))
+            self._process_monitor.start()
+            return process
+        except:
```

Comment:
Is there any chance we can make this except block a bit more specific rather than just a catch all? Can we catch `ThreadError` and/or `RuntimeError`?

Reply:
I realize that the common wisdom is to catch errors specifically, but that only applies when your handling code responds to a particular error. In this case, the handling code is cleanup that should run *whenever* the code in question raises an exception, regardless of which one exactly (since then the contract is: When an exception is raised, no process is left running).

Reply:
OK, fair enough--good point.